lib/FrameManager.js

```diff
@@ -191,11 +191,34 @@ class Frame {
     let remoteObject = await this._rawEvaluate(selector => document.querySelector(selector), selector);
     if (remoteObject.subtype === 'node')
       return new ElementHandle(this._client, remoteObject, this._mouse);
-    helper.releaseObject(this._client, remoteObject);
+    await helper.releaseObject(this._client, remoteObject);
     return null;
   }
 
   /**
+   * @param {string} selector
+   * @return {!Promise<!Array<!ElementHandle>>}
+   */
+  async $$(selector) {
+    const remoteObject = await this._rawEvaluate(selector => Array.from(document.querySelectorAll(selector)), selector);
+    const response = await this._client.send('Runtime.getProperties', {
+      objectId: remoteObject.objectId,
+      ownProperties: true
+    });
+    const properties = response.result;
+    const result = [];
+    const releasePromises = [helper.releaseObject(this._client, remoteObject)];
+    for (const property of properties) {
+      if (property.enumerable && property.value.subtype === 'node')
+        result.push(new ElementHandle(this._client, property.value, this._mouse));
+      else
+        releasePromises.push(helper.releaseObject(this._client, property.value));
```

Comment:
When can we get a non-node result here? If someone overrides querySelectorAll? Maybe we should throw.